libraries/Microsoft.Bot.Builder/BotState.cs

```diff
@@ -112,11 +112,8 @@ public Task ClearStateAsync(ITurnContext turnContext, CancellationToken cancella
                 throw new ArgumentNullException(nameof(turnContext));
             }
 
-            var cachedState = turnContext.TurnState.Get<CachedBotState>(_contextServiceKey);
-            if (cachedState != null)
-            {
-                turnContext.TurnState[_contextServiceKey] = new CachedBotState();
-            }
+            // Explicitly setting the hash will mean IsChanged is always true. And that will force a Save.
+            turnContext.TurnState[_contextServiceKey] = new CachedBotState { Hash = string.Empty };
```

Comment:
This is not clearing out the actual `State` dictionary here though? So `SaveChangesAsync` will see it as changed, but what it will do now (I think) is save whatever values are in the cache as of this turn which may be partially modified and certainly not "cleared". What am I missing? 🤔

Reply:
I was wrong, this will work.